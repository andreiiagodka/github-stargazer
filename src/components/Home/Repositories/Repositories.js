import React from 'react'
import PropTypes from 'prop-types'

import { ListGroup } from 'react-bootstrap'

import Blank from './Blank/Blank'
import Repository from './Repository/Repository'

const Repositories = ({ repositories }) => {
  const fetchedRepositories = repositories.map(repository => {
    return (
      <Repository
        key={repository.id} 
        id={repository.id}
        name={repository.full_name}
        stars={repository.stargazers_count} />
    )
  })

  return ( repositories.length ? <ListGroup>{fetchedRepositories}</ListGroup> : <Blank /> )
}

Repositories.propTypes = {
  repositories: PropTypes.array
}

export default Repositories