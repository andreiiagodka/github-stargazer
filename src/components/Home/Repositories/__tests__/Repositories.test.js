import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Repositories from '../Repositories'

describe('<Repositories /> component', () => {
  it('when repositories are not empty', () => {
    const props = {
      repositories: [
        {
          id: 1,
          full_name: 'Full name',
          stargazers_count: 1
        }
      ]
    }
    const tree = renderer.create(
      <MemoryRouter>
        <Repositories {...props} />
      </MemoryRouter>
    ).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
  
  it('when repositories are empty', () => {
    const props = { repositories: [] }
    const tree = renderer.create(<Repositories {...props} />).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
