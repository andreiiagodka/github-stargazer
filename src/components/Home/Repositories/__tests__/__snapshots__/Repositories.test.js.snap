// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repositories /> component when repositories are empty 1`] = `
<div
  className="row"
>
  <div
    className="text-center col-md-12"
  >
    <p
      className="lead"
    >
      No repositories added
    </p>
  </div>
</div>
`;

exports[`<Repositories /> component when repositories are not empty 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <a
    href="/details/1"
    onClick={[Function]}
  >
    <button
      className="list-group-item list-group-item-action"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="col-md-7"
        >
          Full name
        </div>
        <div
          className="text-right col-md-5"
        >
          1
        </div>
      </div>
    </button>
  </a>
</div>
`;
